### PR TITLE
Initial commit for autogpt-twitter plugin - Twitter API Integrations for AutoGPT using Tweepy!

### PR DESCRIPTION
**Pull request by desojo**
This is the initial commit for autogpt-twitter. View the readme.md located within the autogpt-twitter plugin for installation and setup instructions!

### CODE REVIEW

**Comment by ntindle** [comment]
Looks like we should adjust the scaffolding repo. I think you should just need to replace the twitter plugin’s __init__.py and add the twitter.py file in the same twitter directory

File: plugins/autogpt-twitter/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+black
+isort
+flake8
+pylint
+abstract-singleton
+wheel
+setuptools
+build
+twine
+tweepy
```
Comment (ntindle) [inline plugins/autogpt-twitter/requirements.txt:10]: Update the base repo with this

File: plugins/autogpt-twitter/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# autogpt-twitter 🐣
```
Comment (ntindle) [inline plugins/autogpt-twitter/README.md:1]: Move this to plugins/twitter

File: plugins/autogpt-twitter/src/autogpt_twitter/__init__.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
+"""Twitter API integrations using Tweepy."""
```
Comment (ntindle) [inline plugins/autogpt-twitter/src/autogpt_twitter/__init__.py:1]: Move to plugins/twitter

File: plugins/autogpt-twitter/src/autogpt_twitter/twitter.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
+"""This module contains functions for interacting with the Twitter API."""
```
Comment (ntindle) [inline plugins/autogpt-twitter/src/autogpt_twitter/twitter.py:1]: Move to plugins/twitter

**Comment by desojo** [comment]
I'm a bit confused, do you just want me to rename the  autogpt-twitter directory to twitter, and only include the `__init__.py`, `twitter.py`, `requirements.txt` , and `readme.md` files?

**Comment by ntindle** [comment]
Yes, the rest for code quality coverage etc should use the ones from the root of the repo. I don’t think the example plug-in package is very clear about that is it?

**Comment by desojo** [comment]
Not really, but we're figuring it out as we go!

Don't quote me, but I believe the current example plug-in package is fairly complex with a lot of extra functionality for integrating your own LLM, it's not anywhere near as simple as a plugin for adding in commands to interface with an API.

**Comment by ntindle** [comment]
That makes sense. Part of the goal of getting the twitter one in first is twitter like things are a really common request so having an example of it is really helpfil

**Comment by desojo** [comment]
Alright, what do you think of the changes so far?

File: plugins/twitter/pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+extend-exclude = ""
+
+[tool.isort]
+profile = "black"
+
+[tool.pylint.messages_control]
+disable = "C0330, C0326"
+
+[tool.pylint.format]
+max-line-length = "88"
```
Comment (ntindle) [inline plugins/twitter/pyproject.toml:38]: Don't think we need this one

File: plugins/twitter/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+abstract-singleton
+wheel
+setuptools
+build
+twine
+tweepy
+pandas
+auto_gpt_plugin_template
+python-dotenv
+pathlib
```
Comment (ntindle) [inline plugins/twitter/requirements.txt:14]: Didn't really consider this. Make a new file in the base of the repo called requirements_all.txt with the following contents:

```
# Auto-GPT Plugins Full dependency set
-r requirements.txt

# plugins/twitter
tweepy
```

And include any specific packages you need

**Comment by ntindle** [comment]
The changes are really good! Just a few comments as I figure this all out, sorry!

**Comment by ntindle** [comment]
Wait, sorry again for you being the test pilot, but I think the path is supposed to be src/autogpt_plugins/twitter/__init__.py, etc., for all of these. I forgot we moved them.

**Comment by ntindle** [comment]
Looking way better, See 
> Wait, sorry again for you being the test pilot, but I think the path is supposed to be src/autogpt_plugins/twitter/init.py, etc., for all of these. I forgot we moved them.

They were moved to simplify the imports

**Review by ntindle — CHANGES_REQUESTED** [review]
See comment above

**Review by ntindle — APPROVED** [review]

**Pull request merged**